relay-server/src/actors/relays.rs

```diff
@@ -208,165 +322,41 @@ impl RelayCache {
-    pub relays: HashMap<RelayId, Option<RelayInfo>>,
-}
-
-impl From<PublicKeysResultCompatibility> for GetRelaysResult {
-    fn from(relays_info: PublicKeysResultCompatibility) -> Self {
-        let relays = if relays_info.relays.is_empty() && !relays_info.public_keys.is_empty() {
-            relays_info
-                .public_keys
-                .into_iter()
-                .map(|(id, pk)| (id, pk.map(RelayInfo::new)))
-                .collect()
-        } else {
-            relays_info.relays
-        };
-        Self { relays }
+        if !self.backoff.started() {
+            self.schedule_fetch();
+        }
     }
 }
 
-/// Defines a compatibility format for deserializing relays info that supports
-/// both the old and the new format for relay info
-#[derive(Debug, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct PublicKeysResultCompatibility {
-    /// DEPRECATED. Legacy format only public key info.
-    #[serde(default, rename = "public_keys")]
-    pub public_keys: HashMap<RelayId, Option<PublicKey>>,
-    /// A map from Relay's identifier to its information.
-    #[serde(default)]
-    pub relays: HashMap<RelayId, Option<RelayInfo>>,
-}
+impl Service for RelayCacheService {
+    type Interface = RelayCache;
 
-impl Message for GetRelays {
-    type Result = Result<GetRelaysResult, KeyError>;
-}
+    fn spawn_handler(mut self, mut rx: relay_system::Receiver<Self::Interface>) {
+        tokio::spawn(async move {
+            relay_log::info!("key cache started");
 
-impl UpstreamQuery for GetRelays {
-    type Response = PublicKeysResultCompatibility;
+            loop {
+                tokio::select! {
+                    biased;
 
-    fn method(&self) -> Method {
-        Method::POST
-    }
-
-    fn path(&self) -> Cow<'static, str> {
-        Cow::Borrowed("/api/0/relays/publickeys/")
-    }
-
-    fn priority() -> RequestPriority {
-        RequestPriority::High
-    }
-
-    fn retry() -> bool {
-        false
-    }
-}
-
-impl Handler<GetRelays> for RelayCache {
-    type Result = Response<GetRelaysResult, KeyError>;
-
-    fn handle(&mut self, message: GetRelays, context: &mut Self::Context) -> Self::Result {
-        let mut relays = HashMap::new();
-        let mut futures = Vec::new();
-
-        for id in message.relay_ids {
-            match self.get_or_fetch_info(id, context) {
-                Response::Future(fut) => {
-                    futures.push(fut);
-                }
-                Response::Reply(Ok((id, key))) => {
-                    relays.insert(id, key);
-                }
-                Response::Reply(Err(_)) => {
-                    // Cannot happen
+                    Some(result) = self.fetch_channel.1.recv() => self.handle_fetch_result(result),
+                    () = &mut self.delay => self.fetch_relays(),
+                    Some(message) = rx.recv() => self.get_or_fetch(message.0, message.1),
+                    else => break,
```

Comment:
should this break log a sentry error?  relay is now very broken, some stuff will not happen and we'll need to debug why.  it'd be nice to have a sentry error for this rather than having to look for that info message that says this thing has stopped.

Reply:
This break is in fact dead code and cannot be reached because `self.delay` will always be pending. That we have to change in a follow-up for all services at some point in the future when we revise the shutdown strategy.